services/ui/serverless.yml

```diff
@@ -183,18 +195,17 @@ resources:
           Runtime: cloudfront-js-1.0
         Name: hsts-${self:custom.stage}
     ###############This code block enables logging on waf and sends all logs to s3.##################################
+    WaflogsUploadBucket:
+      Type: AWS::S3::Bucket
+      Properties:
+        BucketName: !Sub ${AWS::AccountId}-${self:service}-${self:custom.stage}-waflogs
     Firehose:
       Type: AWS::KinesisFirehose::DeliveryStream
       Properties:
-        DeliveryStreamName: !Join
-          - ""
-          - - aws-waf-logs-
-            - !Ref AWS::StackName
+        DeliveryStreamName: ${self:custom.firehoseStreamName}
```

Comment:
We set our desired firehosename in a custom variable, so we grab it from there.  The deploy:finalize hook which does the association uses the firehoseName to find the firehose ARN, which is required to make the association.